lib/memo_wise.rb

```diff
@@ -4,8 +4,7 @@
 
 module MemoWise
   def initialize(*values)
-    @_memo_wise = {}
-    @_memo_wise_keys = Hash.new { |h, k| h[k] = [] }
+    @_memo_wise = Hash.new { |h, k| h[k] = {} }
```

Comment:
I also tried a version where this had no default proc and we just created the sub-hash when necessary, but I didn't notice a performance difference and this is a bit easier for the args cases.

Reply:
What if, as an experiment, we don't allocate the `{}` in the `Hash` default proc, so that it doesn't impact the no-args cases?